indexer/services/ender/src/handlers/order-fills/abstract-order-fill-handler.ts

```diff
@@ -325,16 +338,48 @@ export abstract class AbstractOrderFillHandler<T> extends Handler<T> {
     return OrderTable.upsert(orderToCreate, { txId: this.txId });
   }
 
+  /**
+   * The obvious case is if totalFilled >= size, then the order status should always be `FILLED`.
+   * The difficult case is if totalFilled < size after a fill, then we need to keep the following
+   * cases in mind:
+   * - Stateful Orders - All cancelations are onchain events, so the order can be `OPEN` or
+   *   `BEST_EFFORT_CANCELED` if the order is in the CanceledOrdersCache.
+   * - Shortterm FOK - FOK orders can never be `OPEN`, since they don't rest on the orderbook, so
+   *   totalFilled cannot be < size.
+   * - Shortterm IOC - Protocol guarantees that an IOC order will only ever be filled in a single
+   *   block, so status should be `CANCELED`.
+   * - Shortterm Limit & Postonly - If the order is in the CanceledOrdersCache, then it should be
+   *   set to `BEST_EFFORT_CANCELED`, otherwise `OPEN`.
+   * @param isCanceled - if the order is in the CanceledOrderCache, always false for liquidiation
+   * orders
+   */
   protected getOrderStatus(
     isCanceled: boolean,
     size: string,
     totalFilled: string,
+    orderFlags: number,
+    timeInForce: TimeInForce,
   ): OrderStatus {
-    if (isCanceled) {
-      return OrderStatus.BEST_EFFORT_CANCELED;
-    }
-    if (Big(size).lte(totalFilled)) {
+    if (Big(totalFilled).gte(size)) {
       return OrderStatus.FILLED;
+    } else if (orderFlags === ORDER_FLAG_LONG_TERM) {
+      if (isCanceled) {
+        return OrderStatus.BEST_EFFORT_CANCELED;
+      }
+      return OrderStatus.OPEN;
+    } else if (timeInForce === TimeInForce.FOK) {
+      logger.error({
+        at: 'orderFillHandler#getOrderStatus',
+        message: 'FOK orders should never be partially filled',
+        blockHeight: this.block.height,
+        transactionIndex: this.indexerTendermintEvent.transactionIndex,
+        eventIndex: this.indexerTendermintEvent.eventIndex,
+      });
```

Comment:
Is this different behavior than the SQL handler? Do we want to fail to update orders when time in force is FOK and we see a partial fill?

Reply:
I think ideally we would want the SQL handler to mimic the behavior here where it's logging an error, and just seetting the order as `CANCELED`, while it's technically invalid, I just don't want to have the chain halting because of this. I'll add a todo that once logs are added to change the SQL function to log and return `CANCELED`

Reply:
https://linear.app/dydx/issue/IND-439/when-fok-orders-are-partially-filled-log-an-error